frontend/app/features/kanban/components/KanbanBoard.tsx

```diff
@@ -1,21 +1,87 @@
+import { useGetAllTasksQuery } from "@store/api/taskSlice";
+import { useState } from "react";
 import styles from "../styles/kanban.module.css";
+import { filterTasks, TaskFilters } from "../utils/filterTasks";
 import CreateTaskForm from "./CreateTaskForm";
+import TaskFilter from "./TaskFilter";
 
 const KanbanBoard: React.FC = () => {
+  const { data: tasks = [], error, isLoading } = useGetAllTasksQuery();
+  const [filters, setFilters] = useState<TaskFilters>({
+    status: "ALL",
+    createdBy: "ALL",
+    assignedTo: "ALL",
+    dueDate: "ALL",
+    createdAt: "ALL",
+    createdAtOrder: "NEWEST",
+  });
+
+  if (isLoading)
+    return <p className="notification is-info">Loading tasks...</p>;
+  if (error) {
+    console.error("Error loading tasks:", error);
+    return <p className="notification is-danger">Error loading tasks.</p>;
+  }
+
+  const uniqueCreators = [...new Set(tasks.map((task) => task.createdBy))];
+  const uniqueAssignees = [...new Set(tasks.map((task) => task.assignedTo))];
+  const uniqueDueDates = [...new Set(tasks.map((task) => task.dueDate))];
+  const uniqueCreatedAtDates = [...new Set(tasks.map((task) => task.createdAt))];
+
+  const filteredTasks = filterTasks(tasks, filters);
+
   return (
     <div className={styles.kanbanContainer}>
       <div className={styles.sidebar}>
         <CreateTaskForm />
       </div>
 
-      <div className={styles.kanbanBoard}>
-        {["TODO", "IN_PROGRESS", "DONE"].map((status) => (
-          <div key={status} className={styles.column}>
-            <h3 className={styles.columnTitle}>{status.replace("_", " ")}</h3>
-          </div>
-        ))}
+      <TaskFilter
+        uniqueCreators={uniqueCreators}
+        uniqueAssignees={uniqueAssignees}
+        uniqueDueDates={uniqueDueDates}
+        uniqueCreatedAtDates={uniqueCreatedAtDates}
+        onFilterChange={(newFilters) =>
+          setFilters((prev) => ({ ...prev, ...newFilters }))
+        }
+      />
+
+      <div className={`${styles.kanbanBoard} columns is-variable is-4`}>
+        {["TODO", "IN_PROGRESS", "DONE"]
+          .filter(
+            (status) => filters.status === "ALL" || filters.status === status
+          )
+          .map((status) => (
+            <div key={status} className="column is-one-third">
+              <h3 className="title is-4 has-text-centered">
+                {status.replace("_", " ")}
+              </h3>
+              <div className="box has-background-light p-3">
+                {filteredTasks
+                  .filter((task) => task.status === status)
+                  .map((task) => (
+                    <div key={task.id} className="card mb-3">
+                      <header className="card-header">
+                        <p className="card-header-title">{task.title}</p>
+                      </header>
+                      <div className="card-content">
+                        <p>{task.description}</p>
+                        <p className="has-text-grey-light">
```

Comment:
please implement the filters on BE not on FE, the filters are always should be implemented on BE, because there are cases where you want to use pagination, or infinite loader for bigger data (which you can implement later if you want to improve this app), so that it does not take too much time trying to load large data instead fetch only few at first, and load more as users goes, 

in that scenario, FE filter is not efficient, considering you have to filter from all data, but FE only has a proportion of them displayed due to pagination or infinite load.

so, please change this to BE filter instead